ts/src/profile-serializer.ts

```diff
@@ -306,7 +306,7 @@ export function serializeTimeProfile(
         ? generateLabels({node: entry.node, context})
         : context.context;
       if (Object.keys(labels).length > 0) {
-        const values = [1, intervalNanos];
+        const values = unlabelledHits > 0 ? [1, intervalNanos] : [0, 0];
```

Comment:
Is this what avoids emitting wall time for non-JS threads? It feels a bit opaque, can you walk me through it? I'm trying to understand how can you even end up in a branch where you generated labels but `unlabelledHits == 0`? Can `unlabelledHits` drop below zero too?

Reply:
For `(non-JS threads)`, we create a special profile node with zero hits (to mark that it has no associated wall time). Still it has generated labels (`thread name`, ...).
`unlabelledHits` will drop below zero for this sample, but that should not be an issue.

Reply:
I will add a comment (and make sure that `unlabelledHits` does not drop below zero since this can be confusing).

Reply:
Got it – thanks for adding a clarifying comment! I see you also added code to stop decrementing at zero, but I don't think that's really necessary; we can keep it now though.

Reply:
I was wary of the code for the unlabelled samples:
```
    if (unlabelledHits > 0 || unlabelledCpuTime > 0) {
      const labels = generateLabels ? generateLabels({node: entry.node}) : {};
      const values = [unlabelledHits, unlabelledHits * intervalNanos];
```
Even if `unlabelledHits < 0` and `unlabelledCpuTime > 0` should not be possible.
I remove the check to stop decrementing at zero and made the same change as above:
```
const values = unlabelledHits > 0 ? [1, intervalNanos] : [0, 0];
```